### DISCUSSION TITLE
Moving Cloudlog to a new server

### DISCUSSION THREAD

**Discussion by lbismarck** [magicbug/Cloudlog#2242] [General]
Hi all,

I need to move my current Cloudlog installation to a new server as the old one does not allow running cronjobs etc. I am by far no linux or server pro so a little guidance would be very helpful. 
Should I just install a new instance on the new server and then overwrite certain files with the ones I have from the other server? If so which ones exactly in order to have a 100% image on the new server? Or is there a better way?

Thanks and vy 73 de Leo DL2COM

**Comment by magicbug** [comment]
Should be safe enough to copy the files and the database across, you might just need to double check file permissions, make any changes to the database connection info in /application/config/database.php

**Comment by gx1400** [comment]
Can confirm, I did this recently when moving domains on the same server.

Copied cloudlog.blah1.us folder to cloudlog.blah2.us, created new apache2 conf files, loaded up new blah2.us URL and it consistly redirected me back to blah1.us URL

Updated config in cloudlog files to new url, stopped redirecting

Updated my nightly update script, couple of days later noticed that on /debug I did not have write permissions to folders

Went back thru the wiki and found I also needed to 'sudo chmod -R g+rw' not just the chowns.

Fixed

**Comment by lbismarck** [comment]
Thanks a lot for your help! However I am pretty new to server/web administration. I just set the new server up so the LAMP stack is running, no DB created yet. PHP 8.1 is running. I understand Cloulog is compatible as of the latest release with 8.1?

Should I now install a fresh Cloudlog & fresh mysql DB on the new server first and then only copy/overwrite certain files? If so, which ones? 
Or take the entire "public_html" folder and dump it on the new server? how to copy the DB then? Sorry, this feels super silly. 
Thanks again!
![cloudlog-files](https://github.com/magicbug/Cloudlog/assets/1456515/5efba1e2-5370-41dd-8905-c567335d4d5b)

**Comment by phl0** [comment]
> PHP 8.1 is running. I understand Cloulog is compatible as of the latest release with 8.1?

Yepp should work. I guess we fixed most of the bugs.

> Should I now install a fresh Cloudlog & fresh mysql DB on the new server first and then only copy/overwrite certain files? If so, which ones? Or take the entire "public_html" folder and dump it on the new server? how to copy the DB then?

I would copy all the files from public_html to the folder on the new machine. Then do a dump of the DB and import it to the DB server on the new machine. And you should be ready to go.

**Comment by lbismarck** [comment]
@phl0 ok, I did just that (plain copy paste) but the server is throwing a 500. ssl certs have been activated. I created a fresh database and imported the dump from the old installation. the new cloudlog is supposed to be accessible unter the subdomain cloudlog.cqdx.me and before it was www.cqdx.me. the subdomain is already pointing in the correct directory on the new server.

I did change the base URL in the config.php to match the new subdomain

I updated database.php to match the credentials of the new DB.

currently I have PHP 7.3 running to make sure the current version still works (update planned later when it is running).

any hints about which files on the new system now need to be edited would be super helpful. tnx.

**Reply by lbismarck** [reply]
ok, it is working. typical case of - give it some time and it will solve itself - haha. had to adapt the host of the new DB. that fixed it.

**Reply by phl0** [reply]
Ok besides DB connection you probably had to esit application/config/config.php and update the base_url variable.
